RasterDE_BY_GK4_UTM32DirInv.py

```
@@ -0,0 +1,156 @@
+                                                     'NTv2 Grid',
+                                                     options=[i[0] for i in self.grids],
+                                                     defaultValue=0))
+        self.addParameter(QgsProcessingParameterRasterDestination(self.OUTPUT,
+                                                                  'Output'))
+
+    def getConsoleCommands(self, parameters, context, feedback, executing=True):
+        inLayer = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        if inLayer is None:
+            raise QgsProcessingException(self.invalidRasterError(parameters, self.INPUT))
+
+        outFile = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+        self.setOutputValue(self.OUTPUT, outFile)
+
+        direction = self.parameterAsEnum(parameters, self.TRANSF, context)
+        epsg = self.datums[self.parameterAsEnum(parameters, self.CRS, context)][1]
+        grid = self.grids[self.parameterAsEnum(parameters, self.GRID, context)][1]
+
+        found, text = de_transformation(epsg, grid)
+        if not found:
+           raise QgsProcessingException(text)
+
+        arguments = []
+
+        if direction == 0:
+            # Direct transformation
+            arguments.append('-s_srs')
+            arguments.append(text)
+            arguments.append('-t_srs')
+            arguments.append('EPSG:25832')
+        else:
+            # Inverse transformation
+            arguments = ['-s_srs']
+            arguments.append('EPSG:25832')
+            arguments.append('-t_srs')
+            arguments.append(text)
+
+        arguments.append('-multi')
+        arguments.append('-of')
+        arguments.append(QgsRasterFileWriter.driverForExtension(os.path.splitext(outFile)[1]))
+        arguments.append(inLayer.source())
+        arguments.append(outFile)
+
+        gridFile = os.path.join(pluginPath, 'grids', 'BY_KANU.gsb')
+        if not os.path.isfile(gridFile):
```

Better to check and fail early, at the beginning of the algorithm.